index.js

```diff
@@ -100,6 +137,38 @@ class Rewt {
   _generateKeyName() {
     return `${this.options.redisNamespace}:rewt-secret`;
   }
+
+  /**
+   * Cache the secret for as long as the current TTL. While this may be the
+   * incorrect time period, we'll clear the value on any verification errors.
+   *
+   * @param {String} secret The secret to hold on to.
+   * @param {Number} ttl The time to retain the secret in seconds.
+   */
+  _cacheSecret(secret, ttl) {
+    this._cachedSecret = secret;
+    this._clearCachedSecretRef = setTimeout(() => {
```

Comment:
We could also cache the expiration time to avoid needing a timer

Reply:
That's fair, might be a fun add on later.